synapse/module_api/__init__.py

```diff
@@ -266,3 +269,35 @@ async def complete_sso_login_async(
         await self._auth_handler.complete_sso_login(
             registered_user_id, request, client_redirect_url,
         )
+
+    async def room_is_in_public_directory(self, room_id: str) -> bool:
```

Comment:
I do think we need to organise this a bit, which is why I said "make it [PublicRoomsManager] a *property* of the generic ModuleApi".  But I guess it's ok.

Reply:
Ahh, I thought you were referring to its functions.

In the future I can see a sprawling `ModuleApi` with 200+ functions, so I'm in full agreement.